### PR TITLE
Alert on error: shared code

### PR DESCRIPTION
**Pull request by AngeloStavrow**
This PR addresses, but does not close, #204.

It presents an alert to the user when an error is encountered in the "shared" code for the project, with two notable exceptions: the **LocalStoreManager** and the **CollectionListModel**, where thrown errors would not easily be handled at the call site. I didn't want to add refactors of these classes to an already large PR, so for now, I make sure that we crash the app via `fatalError` (for now).

## Alert: no network

<video src="https://user-images.githubusercontent.com/387655/170490639-f4a1a682-c521-4493-8fc3-d46b24214a91.mp4" width="300px" />

## Alert: bad password

<video src="https://user-images.githubusercontent.com/387655/170491351-25e93041-9a25-4bf7-b043-85b32e7bebcb.mp4" width="300px" />

### CODE REVIEW

File: Shared/Account/AccountLoginView.swift [hunk 1]
```diff
@@ -88,13 +88,6 @@ struct AccountLoginView: View {
                 .padding()
             }
         }
-        .alert(isPresented: $model.isPresentingLoginErrorAlert) {
```
Comment (AngeloStavrow) [inline Shared/Account/AccountLoginView.swift:91]: This is handled by the `errorHandling.handle(error:)` call on l80

File: Shared/Account/AccountModel.swift [hunk 2]
```diff
@@ -1,58 +1,6 @@
 import SwiftUI
 import WriteFreely
 
-enum AccountError: Error {
```
Comment (AngeloStavrow) [inline Shared/Account/AccountModel.swift:4]: Moved this to the ErrorConstants file

File: Shared/Extensions/UserDefaults+Extensions.swift [hunk 3]
```diff
@@ -17,17 +17,6 @@ enum WFDefaults {
 
 extension UserDefaults {
 
-    private enum DefaultsError: Error {
```
Comment (AngeloStavrow) [inline Shared/Extensions/UserDefaults+Extensions.swift:20]: Moved this to the ErrorConstants file

File: Shared/Extensions/WriteFreelyModel+Keychain.swift [hunk 4]
```diff
@@ -2,12 +2,6 @@ import Foundation
 
 extension WriteFreelyModel {
 
-    enum WFKeychainError: Error {
```
Comment (AngeloStavrow) [inline Shared/Extensions/WriteFreelyModel+Keychain.swift:5]: Moved this to the ErrorConstants file

File: Shared/Models/WriteFreelyModel.swift [hunk 5]
```diff
@@ -6,28 +6,48 @@ import Network
...
+                #if DEBUG
+                print("  > self.currentError != nil: \(self.currentError != nil)")
+                #endif
+                self.hasError = self.currentError != nil
+                #if DEBUG
+                print("  > hasError is now: \(self.hasError)")
+                #endif
+            }
+        }
+    }
```
Comment (AngeloStavrow) [inline Shared/Models/WriteFreelyModel.swift:34]: Currently we're printing some helpful debug messages to the console when the `currentError` is set; I plan to remove all this when #204 is finalized.

File: Shared/PostCollection/CollectionListModel.swift [hunk 6]
```diff
@@ -19,7 +19,8 @@ class CollectionListModel: NSObject, ObservableObject {
             try collectionsController.performFetch()
             list = collectionsController.fetchedObjects ?? []
         } catch {
-            print("Failed to fetch collections!")
+            // FIXME: Errors cannot be thrown out of the CollectionListView property initializer
+            fatalError(LocalStoreError.couldNotFetchCollections.localizedDescription)
```
Comment (AngeloStavrow) [inline Shared/PostCollection/CollectionListModel.swift:23]: In a separate PR, I'll investigate refactoring this logic for fetching collections out of the CollectionListModel and into a `@FetchRequest` property on the CollectionListView, so we can better handle errors.

File: Shared/PostList/PostListView.swift [hunk 7]
```diff
@@ -86,17 +87,6 @@ struct PostListView: View {
                     Spacer()
                     Text(postCount == 1 ? "\(postCount) post" : "\(postCount) posts")
                         .foregroundColor(.secondary)
-                        .alert(isPresented: $model.isPresentingNetworkErrorAlert, content: {
```
Comment (AngeloStavrow) [inline Shared/PostList/PostListView.swift:89]: This is handled by the `onChange(of:)` modifier on l131

File: Shared/PostList/PostListView.swift [hunk 8]
```diff
@@ -148,18 +148,6 @@ struct PostListView: View {
             ToolbarItemGroup(placement: .primaryAction) {
                 if model.selectedPost != nil {
                     ActivePostToolbarView(activePost: model.selectedPost!)
-                        .alert(isPresented: $model.isPresentingNetworkErrorAlert, content: {
```
Comment (AngeloStavrow) [inline Shared/PostList/PostListView.swift:151]: This is handled by the `onChange(of:)` modifier on l163

**Review by thebaer — APPROVED** [review]
Looks good!

**Pull request merged**